chia/full_node/full_node_api.py

```diff
@@ -46,17 +47,20 @@
 from chia.types.unfinished_block import UnfinishedBlock
 from chia.util.api_decorators import api_request, peer_required, bytes_required, execute_task, reply_type
 from chia.util.full_block_utils import header_block_from_block
-from chia.util.generator_tools import get_block_header
+from chia.util.generator_tools import get_block_header, tx_removals_and_additions
 from chia.util.hash import std_hash
 from chia.util.ints import uint8, uint32, uint64, uint128
 from chia.util.merkle_set import MerkleSet
+from chia.full_node.mempool_check_conditions import get_name_puzzle_conditions
 
 
 class FullNodeAPI:
     full_node: FullNode
+    executor: ThreadPoolExecutor
 
     def __init__(self, full_node: FullNode) -> None:
         self.full_node = full_node
+        self.executor = ThreadPoolExecutor(max_workers=1)
```

Comment:
This could only leverage a single extra thread at this point?  I do think if we do a lot of these things we might prefer a single overall thread pool for the process, but I'm fine with thinking that through later.

Reply:
using a single thread here was a deliberate conservative choice. We perform work on behalf of random wallets on the internet, we want to limit the resources we spend on that.